src/App.jsx

```diff
@@ -14,45 +17,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+const REVERSE_FIELD = false;
+
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort(
+      (goods1, goods2) => {
+        switch (sortField) {
+          case SORT_FIELD_NAME:
+            return goods1.localeCompare(goods2);
+
+          case SORT_FIELD_LENGTH:
+            return goods1.length - goods2.length;
+          default:
+            return 0;
+        }
+      },
+    );
+  }
+
+  if (reverseField) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(REVERSE_FIELD);
```

Comment:
```suggestion
  const [isReversed, setIsReversed = useState(REVERSE_FIELD);
```
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96409563/36a8a88c-a4b7-4d7a-99e2-5d1521d78036)